client/util.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+
+	"golang.org/x/exp/constraints"
+)
+
+const MaxIDPerReq = 25
+
+func BatchAdminClientCall[T constraints.Integer](fn func(context.Context, ...string) (T, error), ids ...string) (T, error) {
```

Comment:
Looks like all the callbacks just return `uint32`? In that case, we don't even need the generic parameter here. Just define it as:

```suggestion
func BatchAdminClientCall(fn func(context.Context, ...string) (uint32, error), ids ...string) (uint32, error) {
```